src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMNavigator.java

```diff
@@ -551,7 +551,7 @@ public void visitSources(SCMSourceObserver observer) throws IOException, Interru
                     null,
                     teamUrl
             ));
-            result.add(new BitbucketTeamMetadataAction(serverUrl, credentials, team.getName()));
+            result.add(new BitbucketTeamMetadataAction(serverUrl, bbCredentials, team.getName()));
```

Comment:
Maybe instead of creating a new class you could modify `BitbucketTeamMetadataAction` to take the needed data and pass that along to `BitbucketAvatarCacheSource` and then `BitbucketAvatarCacheSource` can the look up? 


```suggestion
            result.add(new BitbucketTeamMetadataAction(serverUrl, owner, credentialsId, team.getName()));
```

What do you think?

Reply:
That was my original idea - but i found the BitbucketCredentials class and it already had half the code I needed, seemed like a good fit and easier to pass one thing instead of 3. It is also reusable (though in only one place)

My biggest problem is that I do not understand what this "owner" object is and if it is safe to cache it or if it is even needed. From what I read, it seems usually Run object goes there, but I guess not in this case. I probably need to dig in a bit more

Reply:
@dwnusbaum - I think you have some more knowledge in this area, can you share any insights?

Reply:
I don't quite understand the general change. It's not clear to me why recreating a Jenkins credential is going to make any difference for the reported issue, since the secret value in Jenkins is static as far as I can tell.

As far as `owner`, this needs to be the contextual item that is actually using the credentials (no idea what it is for this plugin or how it is being used). You could add a `transient` cache, but then `owner` would need to be part of the cache key.